scripts.py

```diff
@@ -3031,162 +2974,129 @@ def _explain_coverage_record(self, timestamp, data_source, operation,
         ))
 
 
-class FixInvisibleWorksScript(CollectionInputScript):
+class WhereAreMyBooksScript(CollectionInputScript):
     """Try to figure out why Works aren't showing up.
 
-    This is a common problem on a new installation.
+    This is a common problem on a new installation or when a new collection
+    is being configured.
     """
     def __init__(self, _db=None, output=None, search=None):
         _db = _db or self._db
-        super(FixInvisibleWorksScript, self).__init__(_db)
+        super(WhereAreMyBooksScript, self).__init__(_db)
         self.output = output or sys.stdout
-        self.search = search or ExternalSearchIndex(_db)
-
-    def run(self, cmd_args=None):
-        parsed = self.parse_command_line(self._db, cmd_args=cmd_args)
-        self.do_run(parsed.collections)
-
-    def do_run(self, collections=None):
-        self.check_libraries()
-        if collections:
-            collection_ids = [c.id for c in collections]
+        try:
+            self.search = search or ExternalSearchIndex(_db)
+        except CannotLoadConfiguration, e:
+            self.out("Here's your problem: the search integration is missing or misconfigured.")
+            raise e
 
-        ready = self._db.query(Work).filter(Work.presentation_ready==True)
-        unready = self._db.query(Work).filter(Work.presentation_ready==False)
+    def out(self, s, *args):
+        if not s.endswith("\n"):
+            s += "\n"
+        self.output.write(s % args)
 
-        if collections:
-            ready = ready.join(LicensePool).filter(LicensePool.collection_id.in_(collection_ids))
-            unready = unready.join(LicensePool).filter(LicensePool.collection_id.in_(collection_ids))
-
-        ready_count = ready.count()
-        unready_count = unready.count()
-        self.output.write("%d presentation-ready works.\n" % ready_count)
-        self.output.write("%d works not presentation-ready.\n" % unready_count)
+    def run(self, cmd_args=None):
+        parsed = self.parse_command_line(self._db, cmd_args=cmd_args or [])
 
-        if unready_count > 0:
-            self.output.write(
-                "Attempting to make %d works presentation-ready based on their metadata.\n" % (unready_count)
-            )
-            for work in unready:
-                work.set_presentation_ready_based_on_content(self.search)
```

Comment:
Our treatment of 'presentation-ready' has dramatically changed since we first wrote this script, and it's very unlikely that a work that's not presentation-ready at this point will become presentation-ready if we call this method. However, it is very possible that this code will make the script take a long time to run. I'd rather have the script run quickly. If there's a problem with a lot of non-presentation-ready works, I'd like whoever's having that problem to escalate it to the devs rather than try to fix it by calling `set_presentation_ready_based_on_content`. Either there's a major problem with one of the vendors or there's a bug in one of the collection `Monitor`s.